File: homework1/14_Longest_Common_Prefix.cpp
#include <string>
#include <iostream>
#include <algorithm>
#include <vector>
#include <climits>
using namespace std;
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {

        // Bước 1 tìm từ có length ngắn nhất
        string result = "";
        int length = INT_MAX;
        for (string x : strs){
            if (x.length() <= length) length = x.length();
        }

        // Bước 2
        for ( int i = 0; i < length; i++){
            for (int j = 0; j < strs.size()-1; j++){
                if ( strs[j][i] != strs[j+1][i]){
                    return result;
                }
            }
            result += strs[0][i];
        }
        return result;
    }
};

int main(){
    vector<string> strs = {"flower", "flow", "flight"};
    Solution sol;
    cout << sol.longestCommonPrefix(strs);
}

File: homework1/14_Longest_Common_Prefix.py
from typing import List
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        result = ""
        length = float("inf");

        for x in strs:
            if len(x) <= length: 
                length = len(x)
        for i in range(0, length, 1):
            for j in range(0, len(strs) -1 , 1):
                if strs[j][i] != strs[j+1][i]:
                    return result
            result += strs[0][i]
        return result


def main():
    pass

if __name__ == "main":
    main()